rfcs/team-membership.md

```diff
@@ -0,0 +1,26 @@
+# Team composition
+
+This document aims to formalize the procedures for getting membership in the Rust
+community team.
+
+------
+
+In general, we should encourage participation from community members who are not part of the team,
+as outlined in the [observer policy](observer-policy.md).
+
+In some cases such participating members may have enough involvement that we feel it advatageous to
+include them in the team. Some reasons for doing this may be:
+
+ - Recognition: If someone is especially active in this space, it is worth adding them to the team to recognize them.
+ - "Officializing": For some tasks, like talking with other companies, it is useful to be able to say you are "X of the Rust Community Team"
+
+Most community team activities should be designed such that as much as possible, team membership is not required.
+
+To add someone as a member, they need to be nominated by an existing member on the private community team mailing list.
+Team members can voice approval/objections, and come to consensus on the topic. Consensus is judged by the community team lead.
+In general for team additions this will skew towards there being unanimous approval with at worst minor concerns.
+
+(NOTE: With the advent of subteams this may include relevant subteam leaders for judging consensus)
+
+Voluntarily leaving the team can be done by contacting the community team lead or mailing list. If you wish to re-join
```

Comment:
Re-joining feels like a corner case, but this could also be "if you wish to re-join after a hiatus or something, contact the team lead"

We could also add a procedure for involuntary removal but I think that's the core team's jurisdiction? idk. Didn't find it necessary to add.